src/notebooks_service.py

```diff
@@ -170,18 +170,32 @@ def launch_notebook(user, namespace, project, commit_sha, notebook=None):
         return redirect(notebook_url)
 
     # 1. launch using spawner that checks the access
+    payload = {
+        'branch': request.args.get('branch', 'master'),
+        'commit_sha': commit_sha,
+        'namespace': namespace,
+        'notebook': notebook,
+        'project': project,
+    }
+    if os.environ.get('GITLAB_REGISTRY_SECRET'):
+        payload['image_pull_secrets'] = payload.get('image_pull_secrets', [])
+        payload['image_pull_secrets'].append(
+            os.environ['GITLAB_REGISTRY_SECRET']
+        )
+    app.logger.debug(
+        'POST {prefix}/users/{user[name]}/servers/{server_name}'.format(
+            prefix=auth.api_url, user=user, server_name=server_name
+        )
+    )
+    app.logger.debug(
+        'data = {}'.format(payload)
```

Comment:
probably don't want to log the image registry secret

Reply:
It's just the names of the secret like 'renku-notebooks-registry' so not really critical.

Reply:
ah true